From 7077680339de3e9f197cbee18160309ae2620b11 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 5 Aug 2022 15:14:45 +0200
Subject: [PATCH 0404/1695] plat-stm32mp2: conf: default enable RNG and RNG PTA

Default enable RNG and RNG PTA for STM32MP2 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ee3c0ace8b2b9d5a09cc437198e21462c7c2e144 plat-stm32mp2: conf: default enable RNG and RNG PTA]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I3d2911b5672ca9f86ede7472c0606fe646640a2d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/262337
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index b919ffd25..f8677eaa7 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -76,6 +76,7 @@ CFG_STM32_RIF ?= y
 CFG_STM32_RIFSC ?= y
 CFG_STM32_RISAB ?= y
 CFG_STM32_RISAF ?= y
+CFG_STM32_RNG ?= y
 CFG_STM32_RTC ?= y
 CFG_STM32_SERC ?= y
 CFG_STM32_STGEN ?= y
@@ -96,6 +97,14 @@ CFG_STM32_EARLY_CONSOLE_UART ?= 2
 # Default disable external DT support
 CFG_EXTERNAL_DT ?= n
 
+# Default enable HWRNG PTA support
+CFG_HWRNG_PTA ?= y
+ifeq ($(CFG_HWRNG_PTA),y)
+$(call force,CFG_STM32_RNG,y,Mandated by CFG_HWRNG_PTA)
+$(call force,CFG_WITH_SOFTWARE_PRNG,n,Mandated by CFG_HWRNG_PTA)
+CFG_HWRNG_QUALITY ?= 1024
+endif
+
 # Enable reset control
 ifeq ($(CFG_STM32MP25_RSTCTRL),y)
 $(call force,CFG_DRIVERS_RSTCTRL,y)
-- 
2.47.2

